eng/pipelines/templates/jobs/ci.tests.yml

```diff
@@ -87,7 +87,7 @@ jobs:
         # This will contain things such as heap dumps hprofs if testing hit OutOfMemory errors, log files captured
         # during testing if tests failed, and linting reports.
         - output: pipelineArtifact
-          artifactName: troubleshooting-$(System.StageName)-$(System.JobName)
+          artifactName: troubleshooting-$(System.StageName)-$(System.JobName)-$(System.JobAttempt)
```

Comment:
This _should_ already be appended under the hood by 1es. Are you having issues with it?

Reply:
I don't see job attempt being appended as seen in this retried pipeline run that failed

![image](https://github.com/user-attachments/assets/2f97774d-25ed-4b20-bfdb-8d12fd6762dc)

Reply:
Huh, seems like it then.

@scbedd @weshaggard am I remembering incorrectly here? I thought the 1es publish outputs had solved this under the hood for us.

Reply:
I believe they filed an issue to fix but I don't think they have fixed it yet.

Reply:
For now, I'll add the JobAttempt as this is something that can fail right now